config/routes.rb

```diff
@@ -1,9 +1,24 @@
 Rails.application.routes.draw do
+
   # The priority is based upon order of creation: first created -> highest priority.
   # See how all your routes lay out with "rake routes".
 
   # You can have the root of your site routed with "root"
-  root 'application#hello'
+  root 'spaces#home'
+
+  get 'login', to: 'sessions#new'
+  get 'logout', to: 'sessions#destroy'
+  get 'posts/relevant', to: 'posts#relevant'
+  get 'posts/update_form', to: 'posts#update_form'
+  get 'users/update_form', to: 'users#update_form'
+
+  resources :posts do 
+    member do
+      get 'follow'
```

Comment:
if this is updating data, it should never be a `get` request; it should be `post` or `patch` if it's a create or update action